src/main/java/com/almostreliable/summoningrituals/recipe/AltarRecipe.java

```diff
@@ -56,7 +56,7 @@ public class AltarRecipe implements Recipe<AltarInvWrapper> {
 
     @Override
     public boolean matches(AltarInvWrapper inv, Level level) {
-        if (inv.getCatalyst().isEmpty() || inv.getInputs().isEmpty() || !catalyst.test(inv.getCatalyst())) {
+        if (inv.getCatalyst().isEmpty() || inv.getInputs().size() != this.inputs.size() || !catalyst.test(inv.getCatalyst())) {
```

Comment:
This change will break the crafting logic if you have more items in the Altar than you need since the Altar would require the exact amount of items the recipe has.

Reply:
In that case, I'll just remove the input check, since having items in the altar when a recipe only requires the catalyst plus sacrifices feels awkward. Let me know if that's an issue.